core/async/src/test_loop/event_handler.rs

```diff
@@ -0,0 +1,135 @@
+use std::time::Duration;
+
+use super::{delay_sender::DelaySender, multi_instance::IndexedLoopEventHandler};
+
+/// An event handler registered on a test loop. Each event handler usually
+/// handles only some events, so we will usually have multiple event handlers
+/// registered to cover all event types.
+pub trait LoopEventHandler<Data, Event> {
+    /// Called once, when the loop runs for the first time.
+    fn init(&mut self, _sender: DelaySender<Event>) {}
+
+    /// Handles an event. If this handler indeed handles the event, it should
+    /// return None after handling it. Otherwise, it should return Some with
+    /// the same event that was passed in, so that it can be given to the next
```

Comment:
The convention is to return Result<(),Event> in case the argument is not consumed.
This underlines the fact that the happy path is returning Ok, and Err requires special handling.
Also it will make the compiler warn in case the result is implicitly dropped.

Reply:
Great idea!! I'll change it up!